### PR TITLE
fix: refactor error reporting code sample

### PR DESCRIPTION
**Pull request by minherz**
align code sample for error reporting with code in other languages (b/275730159):

- use the "same" exception type and message
- refactor the method name

### CODE REVIEW

File: error_reporting/snippets/report_exception_test.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 
 def test_error_sends():
-    report_exception.simulate_error()
+    report_exception.report_exception()
```
Comment (rsamborski) [inline error_reporting/snippets/report_exception_test.py:21]: suggestion: This test doesn't seem to `test` anything besides running the code. Probably not in scope of this change, but could we log an issue to extend this test to check if error_reporting actually worked?

**Review by rsamborski — APPROVED** [review]
Approved with a suggestion for future improvement.

File: error_reporting/snippets/report_exception.py [hunk 2]
```diff
@@ -17,13 +17,13 @@
 
 # [START error_reporting_quickstart]
 # [START error_reporting_setup_python]
-def simulate_error():
+def report_exception():
     from google.cloud import error_reporting
 
     client = error_reporting.Client()
     try:
         # simulate calling a method that's not defined
```
Comment (muncus) [inline error_reporting/snippets/report_exception.py:25]: nit: update this comment, since we're using a generic exception now.

**Review by muncus — APPROVED** [review]

File: error_reporting/snippets/report_exception.py (hunk 2 above)
Comment (minherz) [reply error_reporting/snippets/report_exception.py:25]: Reported as #10631

Comment (minherz) [reply error_reporting/snippets/report_exception.py]: remove the comment because the code is simple and self-explanatory.

**Pull request merged**